build.gradle

```diff
@@ -120,8 +120,12 @@ subprojects {
             testCompile deps.assertj
             testCompile deps.junit
             testCompile deps.mockito
+        }
 
-
+        test {
+            // Set maxParallelForks to a large number and let gradle to force it to a the
+            // max-workers number when needed.
+            maxParallelForks =  12
```

Comment:
1. gradle will reduce the number based on the max-worker config. On my mac, the actual number is 4.

2. I suspect not all tests are run concurrently. Even they do, the overall time is dictated by the slowest test. Good question. I am not sure exactly though. The improvement is observable consistently though.

Do you have any objection to this change?